coderd/healthcheck/derphealth/derp.go

```diff
@@ -208,14 +217,15 @@ func (r *NodeReport) Run(ctx context.Context) {
 
 	// We can't exchange messages with the node,
 	if (!r.CanExchangeMessages && !r.Node.STUNOnly) ||
-		// A node may use websockets because `Upgrade: DERP` may be blocked on
-		// the load balancer. This is unhealthy because websockets are slower
-		// than the regular DERP protocol.
-		r.UsesWebsocket ||
 		// The node was marked as STUN compatible but the STUN test failed.
 		r.STUN.Error != nil {
 		r.Healthy = false
 	}
+
+	// A node may use websockets because `Upgrade: DERP` may be blocked on
+	// the load balancer. This is unhealthy because websockets are slower
+	// than the regular DERP protocol.
+	r.Warning = r.UsesWebsocket
```

Comment:
This is the most important logical change. Using WebSockets won't be considered to be a failure anymore.